02.calendar/cal.js

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env node
+
+import minimist from "minimist";
+import dayjs from "dayjs";
+import ja from "dayjs/locale/ja.js";
+
+dayjs.locale(ja);
+const argv = minimist(process.argv.slice(2));
+
+let month = dayjs().format("M");
+let year = dayjs().format("YYYY");
+
+if (argv["m"]) {
+  month = String(argv["m"]);
+  if (argv["y"]) {
+    year = String(argv["y"]);
+  }
+}
+
+const formatted_month_and_year = `${year}-${month}`;
+
+const create_space = (week_number) => {
+  return "   ".repeat(week_number);
+};
+
+const show_calendar = () => {
+  const first_date = dayjs(formatted_month_and_year).startOf("M").format("D");
+  const last_date = dayjs(formatted_month_and_year).endOf("M").format("D");
+
+  for (let date = first_date; date <= last_date; date++) {
+    const display_date = String(date).padStart(2);
+    const date_week_number = dayjs(`${year}-${month}-${date}`).format("d");
+
+    if (date === "1") {
+      process.stdout.write(`${create_space(date_week_number)} 1 `);
+    } else {
+      process.stdout.write(`${display_date} `);
+    }
```

Comment:
```suggestion
    if (date === "1") {
      process.stdout.write(create_space(date_week_number));
    }
    process.stdout.write(`${display_date} `);
```

スペースの出力と日付の出力をきれいに分けた方がコードの理解がしやすそうです

Reply:
なるほど、こちらの方が全然わかりやすいですね😆
余談ですが、後置ifで書こう！と思ったらできませんでした笑

Reply:
後置ifはないので、人によっては代わりにこういう書き方をする場合がありますね。

```javascript
if (date === "1") process.stdout.write(create_space(date_week_number));
```

まあ、可読性があまりよくないので僕自身は滅多に使いませんが😅

Reply:
そんな書き方が...確かに若干読みにくいですが、
参考になります！ありがとうございます🙇‍♂️